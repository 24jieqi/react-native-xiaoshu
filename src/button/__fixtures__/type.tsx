/**
 * title: 按钮类型
 * description: 按钮有五种类型：主按钮（默认）、朦胧按钮、边框按钮、幽灵按钮和链接按钮。
 */

import React, { memo } from 'react'

import {
  Button,
  Card,
  Space,
  Divider,
} from '@fruits-chain/react-native-xiaoshu'

const onPress = () => {
  console.log('点击按钮 => ', new Date().getTime())
}

const ButtonType: React.FC = () => {
  return (
    <Card title="按钮类型" square>
      <Space>
        <Space>
          <Button
            type="primary"
            text="primary 主按钮&点击事件防抖动"
            onPress={onPress}
            onPressDebounceWait={500}
          />
          <Button
            type="hazy"
            text="hazy 朦胧按钮"
            onPress={onPress}
            onPressDebounceWait={300}
          />
          <Button type="outline" text="outline 边框按钮" onPress={onPress} />
          <Button type="ghost" text="ghost 幽灵按钮" onPress={onPress} />
          <Button type="link" text="link 链接按钮" onPress={onPress} />
        </Space>

        <Divider>·</Divider>

        <Space direction="horizontal" wrap>
          <Button type="primary" text="primary" onPress={onPress} />
          <Button type="hazy" text="hazy" onPress={onPress} />
          <Button type="outline" text="outline" onPress={onPress} />
          <Button type="ghost" text="ghost" onPress={onPress} />
          <Button type="link" text="link" onPress={onPress} />
        </Space>
      </Space>
    </Card>
  )
}

export default memo(ButtonType)
