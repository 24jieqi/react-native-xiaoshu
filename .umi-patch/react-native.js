export * from 'react-native-web'

export const InputAccessoryView = () => {}
