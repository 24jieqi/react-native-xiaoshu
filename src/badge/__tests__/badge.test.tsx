import React from 'react'
import type { ViewStyle } from 'react-native'
import { StyleSheet, Text } from 'react-native'
import { create } from 'react-test-renderer'
import type { ReactTestInstance } from 'react-test-renderer'

import Badge from '..'
import { customRender } from '../../__tests__/test-utils'

describe('Badge', () => {
  it('render snapshot', () => {
    const tree = create(<Badge accessibilityLabel="badge" count={6} />).toJSON()

    expect(tree).toMatchSnapshot()
  })

  it('count', () => {
    const { queryByText } = customRender(
      <>
        <Badge count={6} offset={[10, 10]} />
        <Badge count="哈哈" />
        <Badge count={0} />
        <Badge count={12}>
          <Text>Text</Text>
        </Badge>
      </>,
    )

    expect(queryByText('6')).not.toBeNull()
    expect(queryByText('哈哈')).not.toBeNull()
    expect(queryByText('0')).toBeNull()
    expect(queryByText('Text')).not.toBeNull()
  })

  it('color', () => {
    const color = '#f30'
    const { getByLabelText } = customRender(
      <>
        <Badge accessibilityLabel="badge-f30" count={6} color={color} />
        <Badge count="哈哈" />
        <Badge count={0} />
      </>,
    )

    const badgeF30Style = StyleSheet.flatten<ViewStyle>(
      (getByLabelText('badge-f30').children[0] as ReactTestInstance).props
        .style,
    )

    expect(badgeF30Style.backgroundColor).toEqual(color)
  })

  it('showZero', () => {
    const { queryByText } = customRender(<Badge count={0} showZero />)

    expect(queryByText('0')).not.toBeNull()
  })

  it('dot', () => {
    const { queryByText } = customRender(
      <Badge count={13} dot>
        <Text>Text2</Text>
      </Badge>,
    )

    expect(queryByText('13')).toBeNull()
  })

  it('max', () => {
    const { queryByText } = customRender(
      <>
        <Badge count={11} max={20} />
        <Badge count={20} max={20} />
        <Badge count={30} max={20} />
      </>,
    )

    expect(queryByText('11')).not.toBeNull()
    expect(queryByText('20')).not.toBeNull()
    expect(queryByText('30')).toBeNull()
  })
})
