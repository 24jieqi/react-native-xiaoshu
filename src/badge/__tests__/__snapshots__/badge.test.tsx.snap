// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Badge render snapshot 1`] = `
<View
  accessibilityLabel="badge"
  collapsable={false}
>
  <View
    style={
      [
        {
          "borderRadius": 9999,
          "minWidth": 16,
          "paddingHorizontal": 3,
          "paddingVertical": 0,
        },
        {
          "backgroundColor": "#F92F2F",
        },
        null,
        [],
        undefined,
      ]
    }
  >
    <Text
      style={
        [
          {
            "color": "#FFFFFF",
            "fontSize": 10,
            "fontWeight": "bold",
            "height": 16,
            "lineHeight": 16,
            "textAlign": "center",
          },
          undefined,
        ]
      }
    >
      6
    </Text>
  </View>
</View>
`;
